HW0/P5/P5.txt

```diff
@@ -1 +1,25 @@
-# Your answers here.
+By: Matheus C. Fernandes
+
+Question 1: 255 seconds
+	If one cashier were counting the add the totals of 256 bags it would take 255 seconds because the first count 1+1=2 would count as one secong and every count thereafter would also count as one second 
+
+Question 2: 34 seconds
+	Yes, assuming they imidiatelly divide up the work evenly, they can deivide the 256 bags by 8 workers. Each worker than has to count 32 bags which would take 31 seconds to count. Then 4 workers can add up the other bags, taking 1 additional second. Then 2 workers take the sum of 4 people, taking 1 additional second. Finally, the 1 worker adds the final sum of two people, during the last 1 second.
+
+Question 3: 8 seconds
+	Assuming there is no communication time between workers, it would take 8 seconds. This starts with having 256 workers and taking each second to add each of the half of the workers namely it takes log2(256)=8.
+	
+Question 4: log2(N), please find attached P5.png
+	It would take infinite number of employees log2(N) seconds to count N bags.
+
+Question 5: 15 seconds
+	Every time there is a sum between two numbers there is one additional second, except for the first case. 
+
+Question 6: **this part was done in colaboration with Thibaut Perol**
+	(each cashier comes and takes 1 bag): 230 seconds
+		picture attached 'P6_1bagperworker-8bags.jpg,P6_1bagperworker-16bags.jpg'
```

Comment:
The strategy was that we looked at the combinations of workers. The graph represents merges in one color and counts in another color. Given the particular amount of workers and what they can handle, we drew up different schemes where the work could be handed off to each worker. Each increment of time (flop) is represented by the number inside the node in the graph. Therefore the most efficient way we could think of, is drawn through the nodes and through the connections. This, by nature of the problem, is difficult to explain in words. That is why we drew up the diagram and made is as clear as we could through the legend and the numbering of the nodes.